tools/test/toolchains/test_toolchains.py

```diff
@@ -40,6 +43,58 @@ def test_arm_minimal_printf(self):
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_micro_obj.flags["common"])
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_c6_obj.flags["common"])
 
+    def test_arm_default_lib(self):
+        """Test that linker flags are correctly added to an instance of ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.supported_c_libs = {"arm": ["small"]}
+        mock_target.default_lib = "sMALL"
+        mock_target.default_toolchain = "ARM"
+        mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5", "ARMC6"]
+        arm_std_obj = ARM_STD(mock_target)
+        arm_micro_obj = ARM_MICRO(mock_target)
+
+        mock_target.default_toolchain = "ARMC6"
+        arm_c6_obj = ARMC6(mock_target)
+
+        self.assertIn("-D__MICROLIB", arm_std_obj.flags["common"])
+        self.assertIn("-D__MICROLIB", arm_micro_obj.flags["common"])
+        self.assertIn("-D__MICROLIB", arm_c6_obj.flags["common"])
+
+        self.assertIn("--library_type=microlib", arm_std_obj.flags["ld"])
+        self.assertIn("--library_type=microlib", arm_micro_obj.flags["ld"])
+        self.assertIn("--library_type=microlib", arm_c6_obj.flags["ld"])
+
+        self.assertIn("-Wl,--library_type=microlib", arm_c6_obj.flags["cxx"])
+        self.assertIn("--library_type=microlib", arm_c6_obj.flags["asm"])
+
+    def test_arm_default_lib_std_exception(self):
+        """Test that an exception raised if the default_lib is std but it is not listed in supported_c_libs."""
```

Comment:
```suggestion
        """Test that an exception is raised if the std C library is not supported for a target on the ARM toolchain."""
```
Use the same form elsewhere.